pkg/operator/k8sutil/node.go

```diff
@@ -279,3 +281,62 @@ func NodeIsInRookNodeList(targetNodeName string, rookNodes []rookalpha.Node) boo
 	}
 	return false
 }
+
+// AddNodeAffinity will return v1.NodeAffinity or error
+func AddNodeAffinity(nodeAffinity string) (*v1.NodeAffinity, error) {
+	newNodeAffinity := &v1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+			NodeSelectorTerms: []v1.NodeSelectorTerm{
+				{},
+			},
+		},
+	}
+	nodeLabels := strings.Split(nodeAffinity, ";")
+	// For each label in 'nodeLabels', retrieve (key,value) pair and create nodeAffinity
+	// '=' separates key from values
+	// ',' separates values
+	for _, nodeLabel := range nodeLabels {
+		// If tmpNodeLabel is an array of length > 1
+		// [0] is Key and [1] is comma separated values
+		tmpNodeLabel := strings.Split(nodeLabel, "=")
+		if len(tmpNodeLabel) > 1 {
+			nodeLabelKey := tmpNodeLabel[0]
+			tmpNodeLabelValue := tmpNodeLabel[1]
+			nodeLabelValues := strings.Split(tmpNodeLabelValue, ",")
+			if nodeLabelKey != "" && len(nodeLabelValues) > 0 {
+				err := validation.IsQualifiedName(nodeLabelKey)
+				if err != nil {
```

Comment:
`IsQualifiedName` returns a list of strings, so probably better to check for `len()` of the errors list.

Reply:
Documentation says it will return an `empty list (or nil)` if there are no errors. So I thought checking for `nil` is acceptable.

Reply:
Right, my bad. Empty slice is "equal" to nil, see https://stackoverflow.com/a/44305910.
In the function it will always return the `errs` slice which may or not be empty.

No need to change the code.